### PR TITLE
refactor: Move `AppAccess` validation out of EC2 pattern file

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

Part of an ongoing effort to reduce the LOC in `ec2-app` (see #1105 and #1108 for previous work).

This is a no-op in terms of functionality. Hopefully it's as clear, if not more clear, to understand as before.

This change moves some CIDR validation logic away into it's own file for increased readability and testing.

## How to test

CI should continue to pass.

## How can we measure success?

A simpler codebase.

## Have we considered potential risks?

n/a

## Checklist

- [ ] I have listed any breaking changes, along with a migration path [^1]
- [ ] I have updated the documentation as required for the described changes [^2]

[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?

### CODE REVIEW

File: src/patterns/ec2-app/base.test.ts [hunk 1]
```diff
@@ -389,7 +389,9 @@ describe("the GuEC2App pattern", function () {
           monitoringConfiguration: { noMonitoring: true },
           userData: "",
         })
-    ).toThrowError();
```
Comment (akash1810) [inline src/patterns/ec2-app/base.test.ts:392]: Should always test the message being thrown, else something else could have been throwing!

File: src/patterns/ec2-app/base.ts [hunk 2]
```diff
@@ -128,25 +128,6 @@ export interface GuEc2AppProps extends AppIdentity {
...
-  Please either remove the public CIDR range or use the RESTRICTED access scope.`);
-
-function validateRestrictedCidrRanges(access: RestrictedAccess) {
-  const openToWorld = access.cidrRanges.map((range) => range.uniqueId).includes("0.0.0.0/0");
-  if (openToWorld) throw OpenApplicationCidrError;
-}
-
-function validateInternalCidrRanges(access: InternalAccess) {
-  const publicCidrRanges = access.cidrRanges.filter((range) => !range.uniqueId.startsWith("10."));
-  const containsPublicIps = publicCidrRanges.length > 0;
```
Comment (akash1810) [inline src/patterns/ec2-app/base.ts:146]: re-written as [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) for brevity

File: src/types/access.ts [hunk 3]
```diff
@@ -56,3 +57,44 @@ export interface InternalAccess extends Access {
...
+   *
+   * @throws `Error` when CIDR rules are invalid
+   */
+  validate: (access: AppAccess) => {
+    switch (access.scope) {
+      case AccessScope.RESTRICTED: {
+        const ranges = access.cidrRanges.map((_) => _.uniqueId);
+        const rangesString = ranges.join(", ");
+
+        if (ranges.includes(Peer.anyIpv4().uniqueId) || ranges.includes(Peer.anyIpv6().uniqueId)) {
```
Comment (jacobwinch) [inline src/types/access.ts:77]: I think this PR is also adding IPv6 support to this check? 💯

**Review by jacobwinch — APPROVED** [review]

File: src/types/access.ts (hunk 3 above)
Comment (akash1810) [reply src/types/access.ts:77]: Oh. Yes. I guess this is _not_ a no-op after all then! Happy to remove IPv6 stuff or update the PR as `fix:` to create a release.

Comment (jacobwinch) [reply src/types/access.ts:77]: I think it's fine to include it here, I don't think many people will be specifying IPv6 ranges yet! Feel free to update to `fix` if you want to generate a release though!

Comment (akash1810) [reply src/types/access.ts:77]: A new release for this is most likely to just be noise. Will leave as is.

**Pull request merged**